src/text/text_writer.rs

```diff
@@ -0,0 +1,161 @@
+use crate::raw_symbol_token_ref::{AsRawSymbolTokenRef, RawSymbolTokenRef};
+use crate::result::{illegal_operation, IonResult};
+use crate::text::raw_text_writer::RawTextWriter;
+use crate::types::decimal::Decimal;
+use crate::types::timestamp::Timestamp;
+use crate::writer::Writer;
+use crate::IonType;
+use crate::SymbolTable;
+use delegate::delegate;
+use std::io::Write;
+
+/**
+ * An application-level text Ion writer. This writer manages a symbol table and so can convert
+ * symbol IDs to their corresponding text. However, unlike the BinaryWriter, it is capable of writing
+ * text to the output stream without first adding it to the symbol table.
+ */
+pub struct TextWriter<W: Write> {
+    raw_writer: RawTextWriter<W>,
+    symbol_table: SymbolTable,
+}
+
+impl<W: Write> TextWriter<W> {
+    pub fn new(raw_writer: RawTextWriter<W>) -> Self {
+        TextWriter {
+            raw_writer,
+            symbol_table: SymbolTable::new(),
+        }
+    }
+}
+
+impl<W: Write> Writer for TextWriter<W> {
+    fn supports_text_symbol_tokens(&self) -> bool {
+        true
```

Comment:
Should we add a comment around the return value here same as the `RawBinaryWriter`?

Reply:
Added. 👍